Source/SwiftLintFramework/Protocols/Rule.swift

```diff
@@ -32,33 +32,46 @@ public protocol ConfigurableRule: Rule {
     func isEqualTo(rule: ConfigurableRule) -> Bool
 }
 
-public protocol ParameterizedRule: ConfigurableRule {
-    typealias ParameterType: Equatable
-    init(parameters: [RuleParameter<ParameterType>])
-    var parameters: [RuleParameter<ParameterType>] { get }
+public protocol ViolationLevelRule: ConfigurableRule {
+    var warning: RuleParameter<Int> { get set }
```

Comment:
I think you do. `init?(config:)` calls `init()` first to load the default values for a `Rule` before overwriting those values with something that might be in the `config`. This is important to accommodate configurations that aren't fully specified (e.g. `variable_name_min_length: 3`). Because it is a delegating initializer, this needs to be done before you access `self`. The only way around this would be to make `init?(config:)` non-delegating. Which means we can't support both configured values and default values at the same time, it becomes an either/or. At least that is my thinking on it. Could be wrong.